frontend/apps/filemanager/filemanagerutil.lua

```diff
@@ -138,34 +130,64 @@ end
 
 -- Generate "Reset" file dialog button
 function filemanagerutil.genResetSettingsButton(file, caller_callback, button_disabled)
+    file = ffiutil.realpath(file)
+    local sidecar_file = DocSettings:hasSidecarFile(file)
+    local custom_cover_file = DocSettings:findCoverFile(file)
+    local custom_metadata_file = DocSettings:getCustomMetadataFile(file)
     return {
         text = _("Reset"),
         id = "reset", -- used by covermenu
-        enabled = (not button_disabled and DocSettings:hasSidecarFile(ffiutil.realpath(file))) and true or false,
+        enabled = (not button_disabled and (sidecar_file or custom_metadata_file or custom_cover_file)) and true or false,
         callback = function()
+            local CheckButton = require("ui/widget/checkbutton")
             local ConfirmBox = require("ui/widget/confirmbox")
+            local check_button_settings, check_button_cover, check_button_metadata
             local confirmbox = ConfirmBox:new{
                 text = T(_("Reset this document?") .. "\n\n%1\n\n" ..
-                    _("Document progress, settings, bookmarks, highlights, notes and custom cover image will be permanently lost."),
+                         _("Resetted information will be permanently lost."),
                     BD.filepath(file)),
                 ok_text = _("Reset"),
                 ok_callback = function()
-                    local custom_metadata_purged = filemanagerutil.purgeSettings(file)
-                    if custom_metadata_purged then -- refresh coverbrowser cached book info
-                        local FileManager = require("apps/filemanager/filemanager")
-                        local ui = FileManager.instance
-                        if not ui then
-                            local ReaderUI = require("apps/reader/readerui")
-                            ui = ReaderUI.instance
-                        end
+                    local data_to_purge = {
+                        doc_settings         = check_button_settings.checked,
+                        custom_cover_file    = check_button_cover.checked and custom_cover_file,
+                        custom_metadata_file = check_button_metadata.checked and custom_metadata_file,
```

Comment:
There is not .old (backup) of these, right? So we can pass directly the files we have found, right?
But :purge() will find them again if we pass no arg, right?
(Feels a bit odd that 2 places do the same work, but ok.)

Reply:
Right.
We must do this work to enable/disable the ConfirmBox checkbuttons, prior to purge.

Reply:
Ok, but we could just pass `custom_cover_file=true` and `custom_metadata_file=true` (booleans, like the 3rd one `doc_settings = true` is, for consistency), and let :purge() do again its normal work of finding them (again, yes, but this may not need that optimisation).

Reply:
I think we should avoid double scanning the drive to find custom cover and metadata files again.

Reply:
OK, not insisting :) as all this is quite localized and used only by these 2 modules.
(For a cleaner API, we would accept custom_cover_file=true (so all 3 keys _can be_ booleans), and when it is a boolean, find the filepath - but when it is a string, assume it's the filepath that has already been searched and found, and use that as an optimization.)